src/Shape/Whisker.js

```diff
@@ -0,0 +1,257 @@
+import {select} from "d3-selection";
+
+import {accessor, assign, BaseClass, configPrep, constant, elem} from "d3plus-common";
+
+import Circle from "./Circle";
+import Line from "./Line";
+import Rect from "./Rect";
+
+/**
+    @class Whisker
+    @extends BaseClass
+    @desc Creates SVG lines based on an array of data.
+*/
+export default class Whisker extends BaseClass {
+
+  /**
+      @memberof Whisker
+      @desc Invoked when creating a new class instance, and overrides any default parameters inherited from BaseClass.
+      @private
+  */
+  constructor() {
+
+    super();
+
+    this._endpoint = constant("Circle");
+    this._endpointConfig = {
+      Circle: {
+        fill: d => d.fill,
+        r: d => d.r
+      },
+      Rect: {
+        fill: d => d.fill,
+        height: d => d.height,
+        stroke: constant("black"),
+        strokeWidth: constant(1),
+        width: d => d.width
+      }
+    };
+    this._length = accessor("length", 25);
+    this._lineConfig = {
+      stroke: constant("black"),
+      strokeWidth: constant(1)
+    };
+    this._name = "Whisker";
+    this._orient = accessor("orient", "top");
+    this._shapes = [];
+    this._x = accessor("x", 0);
+    this._y = accessor("y", 0);
+
+  }
+
+  /**
+      @memberof Box
+      @desc Filters/manipulates the data array before binding each point to an SVG group.
+      @param {Array} [*data* = the data array to be filtered]
+      @private
+  */
+  _whiskerData(data) {
+    this._lineData = [];
+    this._circleData = [];
+    this._rectData = [];
+    data.forEach((d, i) => {
+      const lineCoordinates1 = {id: i, x: d.x, y: d.y};
+      this._lineData.push(lineCoordinates1);
+
+      if (d.orient === "top") {
+        d.x2 = d.x;
+        d.y2 = d.y - d.length;
+      }
+      else if (d.orient === "bottom") {
+        d.x2 = d.x;
+        d.y2 = d.y + d.length;
+      }
+      else if (d.orient === "left") {
+        d.x2 = d.x - d.length;
+        d.y2 = d.y;
+      }
+      else if (d.orient === "right") {
+        d.x2 = d.x + d.length;
+        d.y2 = d.y;
+      }
+      const lineCoordinates2 = {id: i, x: d.x2, y: d.y2};
+      this._lineData.push(lineCoordinates2);
+
+      if (d.endpoint === undefined) d.endpoint = this._endpoint();
```

Comment:
I'm getting an error on my requested example code because you are not passing the data point to this function. EVERY user function must be passed both the data point (usually `d`) and that data point's index (usually `i`) as their 2 arguments.